src/runtime/server/api/session.get.ts

```diff
@@ -1,10 +1,13 @@
 import { eventHandler } from 'h3'
-import { requireUserSession, sessionHooks } from '../utils/session'
+import { getUserSession, sessionHooks } from '../utils/session'
+import type { UserSessionRequired } from '#auth-utils'
 
 export default eventHandler(async (event) => {
-  const session = await requireUserSession(event)
+  const session = await getUserSession(event)
 
-  await sessionHooks.callHookParallel('fetch', session, event)
+  if (session.user) {
+    await sessionHooks.callHookParallel('fetch', session as UserSessionRequired, event)
+  }
```

Comment:
Is there a reason for requiring that `user` is defined on the session before running the `fetch` hook? This seems to be a breaking change, compared to the previous implementation. For my use case i would like the `user` property to be populated dynamically by a session hook based on other information in the session, but this change prevents that use case.